simulation/traffic_simulator/src/utils/distance.cpp

```diff
@@ -311,5 +311,39 @@ auto distanceToStopLine(
     return spline.getCollisionPointIn2D(polygon);
   }
 }
+
+auto distanceToSpline(
+  const geometry_msgs::msg::Pose & map_pose,
+  const traffic_simulator_msgs::msg::BoundingBox & bounding_box,
+  const math::geometry::CatmullRomSplineInterface & spline, const double s_reference) -> double
+{
+  constexpr double distance_accuracy{0.05};
+
+  const auto bounding_box_map_points =
+    math::geometry::transformPoints(map_pose, math::geometry::getPointsFromBbox(bounding_box));
+  const auto bounding_box_diagonal_length =
+    math::geometry::getDistance(bounding_box_map_points[0], bounding_box_map_points[2]);
+
+  /// @note it may be a good idea to develop spline.getSquaredDistanceIn2D(point, s_start, s_end);
+  std::vector<double> distances;
+  for (const auto & point : bounding_box_map_points) {
+    auto s_start = s_reference - bounding_box_diagonal_length / 2;
+    auto s_end = s_reference + bounding_box_diagonal_length / 2;
+    auto s_start_distance = spline.getSquaredDistanceIn2D(point, s_start);
+    auto s_end_distance = spline.getSquaredDistanceIn2D(point, s_end);
+
+    while (std::abs(s_start - s_end) > distance_accuracy) {
+      if (s_start_distance > s_end_distance) {
+        s_start += (s_end - s_start) / 2;
+        s_start_distance = spline.getSquaredDistanceIn2D(point, s_start);
+      } else {
+        s_end -= (s_end - s_start) / 2;
+        s_end_distance = spline.getSquaredDistanceIn2D(point, s_end);
+      }
```

Comment:
Isn't this a better way to convey your intention?

```suggestion
      double s_mid = std::midpoint(s_start, s_end);
      double s_mid_distance = spline.getSquaredDistanceIn2D(point, s_mid);
      if (s_start_distance > s_end_distance) {
        s_start = s_mid;
        s_start_distance = s_mid_distance;
      } else {
        s_end = s_mid;
        s_end_distance = s_mid_distance;
      }
```

Reply:
I like your proposition. Applied.

Reply:
I haven't noticed that std::midpoint is since c++20 not 17, and it will not build. Maybe I will just change it it to
`double s_mid = s_start + (s_end - s_start) / 2;` ?

Reply:
Oh, sorry for suggesting code with C++20.
`double s_mid = s_start + (s_end - s_start) / 2;` world be fine.

Reply:
done